### PR TITLE
Esp32c6

### PR DESCRIPTION
**Pull request by davidqge**
Fixed several crashes on esp32c6.
Typing is working now, except Vial.

### CODE REVIEW

File: examples/use_config/esp32c6_ble/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# esp32c3 BLE example
+
+To run this example, you should have latest Rust in **esp** channel and `esp-idf` toolchain installed. The full instruction of installing `esp-idf` toolchain can be found [here](https://docs.esp-rs.org/book/installation/index.html) and [here](https://docs.esp-rs.org/std-training/02_2_software.html)
+
+To run the example, make sure that you have esp-idf environment, `ldproxy` and `espflash` installed correctly. Then, run 
+
+```
+cd examples/use_rust/esp32c3_ble
```
Comment (HaoboGu) [inline examples/use_config/esp32c6_ble/README.md]:
```suggestion
cd examples/use_rust/esp32c6_ble
```

**Comment by HaoboGu** [comment]
I checked the BLE report writing, it seems current implementation has a bug when serializing the report:

https://github.com/HaoboGu/rmk/blob/main/rmk/src/ble/esp/server.rs#L31-L38
```diff
async fn write_serialize<IR: AsInputReport>(&mut self, r: &IR) -> Result<(), HidError> {
        use ssmarshal::serialize;
        let mut buf: [u8; 32] = [0; 32];
-       match serialize(&mut buf, &r) {
+       match serialize(&mut buf, r) {
            Ok(n) => self.write(&buf[0..n]).await,
            Err(_) => Err(HidError::ReportSerializeError),
        }
    }

```

Could you please try this fix?

**Comment by HaoboGu** [comment]
Please also add ci for esp32c6 to `.github/workflows/build esp.yml`

**Comment by davidqge** [comment]
Yes, that fixed the crash too.
Added ci, and changed readme.

**Comment by HaoboGu** [comment]
LGTM, thanks a lot!

**Pull request merged**